src/main/java/org/sefglobal/scholarx/service/MentorService.java

```diff
@@ -40,4 +52,44 @@ public Mentor updateState(long id, EnrolmentState enrolmentState)
         optionalMentor.get().setState(enrolmentState);
         return mentorRepository.save(optionalMentor.get());
     }
+
+    /**
+     * Create new {@link Mentee}
+     *
+     * @param profileId which is the Profile id of the applying user's {@link Profile}
+     * @param mentorId  which is mentor id for the requesting {@link Mentor}
+     * @param mentee    which holds the data to be added
+     * @return the created {@link Mentee}
+     * @throws ResourceNotFoundException is thrown if the applying {@link Mentor} doesn't exist
+     * @throws ResourceNotFoundException is thrown if the applying user's {@link Profile} doesn't exist
+     * @throws BadRequestException is thrown if the applying {@link Mentor} is not approved
+     */
+    public Mentee applyAsMentee(long mentorId, long profileId, Mentee mentee) throws ResourceNotFoundException, BadRequestException {
+        Optional<Mentor> optionalMentor = mentorRepository.findById(mentorId);
+        if (!optionalMentor.isPresent()) {
+            String msg = "Error, Unable to apply as a mentee. " +
+                         "Mentor with id: " + mentorId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+        if (!optionalMentor.get().getState().equals(EnrolmentState.APPROVED)) {
+            String msg = "Error, Unable to apply as a mentee. " +
+                         "Mentor with id: " + mentorId + " is not an approved mentor.";
```

Comment:
In case mentor is in `REMOVED`  @state
```suggestion
                         "Mentor with id: " + mentorId + " is not in applicable state.";
```

Reply:
Updated